docs/source/app-dev/bindings-java/quickstart/template-root/src/main/java/com/daml/quickstart/iou/IouMain.java

```diff
@@ -65,23 +60,21 @@ public static void main(String[] args) {
         .getActiveContracts(ContractUtil.of(Iou.COMPANION), Collections.singleton(party), true)
         .blockingForEach(
             activeContracts -> {
-              activeContracts
-                  .getOffset()
-                  .ifPresent(offset -> acsOffset.set(new LedgerOffset.Absolute(offset)));
-              activeContracts
-                  .getContracts()
-                  .forEach(
-                      contract -> {
-                        long id = idCounter.getAndIncrement();
-                        contracts.put(id, contract.data);
-                        idMap.put(id, contract.id);
-                      });
+              activeContracts.offset.ifPresent(
+                  offset -> acsOffset.set(new LedgerOffset.Absolute(offset)));
+              activeContracts.activeContracts.forEach(
+                  contract -> {
+                    long id = idCounter.getAndIncrement();
+                    contracts.put(id, contract.data);
+                    idMap.put(id, contract.id);
+                  });
             });
 
     Disposable ignore =
         client
             .getTransactionsClient()
-            .getTransactions(acsOffset.get(), iouFilter, true)
+            .getTransactions(
+                ContractUtil.of(Iou.COMPANION), acsOffset.get(), Collections.singleton(party), true)
```

Comment:
If the Companion would implement a ContractFilter interface, you could do 

```
.getTransactions(Iou.COMPANION, acsOffset.get(), Collections.singleton(party), true)
```
?

Reply:
Or similar to `valueDecoder` on `Iou` you could have a `contractFilter` and do 
```
getTransactions(Iou.contractFilter(), acsOffset.get(), Collections.singleton(party), true)
```
?

Reply:
Even better, on `Iou.Contract`, so it's `Iou.Contract.filter()` here:
```
getTransactions(Iou.Contract.filter(), acsOffset.get(), Collections.singleton(party), true)
```

Reply:
As generated Interface class doesn't have a static Contract Type in it. So it can not do `IIou.Contract.filter()` May be we can have it in the parent class? which is your previous comment:
```
getTransactions(Iou.contractFilter(), acsOffset.get(), Collections.singleton(party), true)
```
so that we can do `Iou.contractFilter()` for template class and `IIou.contractFilter()` for interface?